helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -639,14 +639,20 @@ private void subscribeForChanges(NotificationContext.Type callbackType, String p
             }
           }
         } catch (ZkNoNodeException | HelixMetaDataAccessException e) {
-          logger.warn(
-              "fail to subscribe child/data change. path: " + path + ", listener: " + _listener, e);
+          // TODO: avoid calling getChildren for path that does not exist
+          if (_changeType == CUSTOMIZED_STATE_ROOT) {
+            logger.warn("Failed to subscribe child/data change. path: {}, listener: {}", path,
+                _listener);
+          } else {
+            logger.warn("Failed to subscribe child/data change. path: {}, listener: {}", path,
+                _listener, e);
```

Comment:
Can we just merge these two conditions? There's practically no difference.

Reply:
@narendly tried to remove duplicates. However, this will make the log and stacktrace to be in two entries. Won't look good.